Unity.gitignore

```diff
@@ -1,36 +1,37 @@
-[Ll]ibrary/
-[Tt]emp/
-[Oo]bj/
-[Bb]uild/
-[Bb]uilds/
-Assets/AssetStoreTools*
+**/[Ll]ibrary/
```

Comment:
We've gone through multiple changes to get this pattern right so I'm hesitant to do this again:

 - for the longest time we had a4661a2d39f9cd733ff616dbcc2918f959828d76 which was `/[Ll]ibrary/` to ignore just the root folder
 - then e0c2baea55e9537dbc50e834a090da34c4229841 in #2502 changed it to `*/[Ll]ibrary/` 
 - then dc0ca6d8d5e61dae1bd051f9a99c1414890069f6 in #2532 changed this to `[Ll]ibrary/`  

I'd love to figure out the right pattern here once and for all, rather than endlessly churning these patterns. cc @matzar who has contributed to recent PRs.

Reply:
@shiftkey at the time I've tested the current .gitignore (`[Ll]ibrary/`) with git BASH v1.9 and v2.* and it worked fine for the ignored folders. The reason for creating a wildcard (`*/`) was that git BASH v1.9 did not seem to work with `/[Ll]ibrary/` but it might have been the case that I did not have the full admin rights.  But again `[Ll]ibrary/` seems to work with all the vesrions.

If you were to ask me what's the best practice for defining ignore folders and extension files in .gitignore that would be: 

Folders: 
`[Nn]ameOfIgnoredFolder/` 
`.nameOfIgnoredFolder/` 

Extnesion files: 
`*.extensionOfIgnoredFiles` 

I believe [30d40a](https://github.com/github/gitignore/pull/2850/commits/30d40a225ac122684ce8f8498d923be15e54817a) is unnecessary